services/src/main/java/org/keycloak/forms/login/freemarker/FreeMarkerLoginFormsProvider.java

```diff
@@ -489,7 +489,7 @@ protected void createCommonAttributes(Theme theme, Locale locale, Properties mes
             IdentityProviderBean idpBean = new IdentityProviderBean(session, realm, baseUriWithCodeAndClientId, context);
 
             if (Profile.isFeatureEnabled(Feature.ORGANIZATION) && realm.isOrganizationsEnabled()) {
-                idpBean = new OrganizationAwareIdentityProviderBean(idpBean);
+                idpBean = new OrganizationAwareIdentityProviderBean(idpBean, user);
```

Comment:
At this point, the organization should have been already resolved during the authentication flow. The forms are only created by authenticators and in the organization authenticator we already took care of resolving the organization based on the user that is authenticating (or during a challenge, so no user at all).

I would rather revert those changes and keep the authenticator as the single source of truth for pushing the organization that forms should be using when rendering pages.

Reply:
Yeah thanks for the suggestion. I checked it, and it works without those changes outside `ResolveOrganization()`